### ISSUE TITLE
When using SFTP in Dolphin, Bitwarden's SSH agent constantly asks for Authorization

### ISSUE THREAD

**Issue by Pixaurora** [bitwarden/clients#12520]
### Steps To Reproduce

1. Launch Dolphin with the environment variable `SSH_AUTH_SOCK~/.bitwarden-ssh-agent.sock` set.
2. Connect to a remote machine using `sftp://[ip]/[folder]`
3. Authorize the SSH request.
4. Click around to more folders.

### Expected Result

Bitwarden should ask for authorization once and then allow you to click around to multiple folders freely.

### Actual Result

Because Dolphin makes multiple separate SFTP requests rather than using a single session, each request it makes needs to be authorized separately and is quite annoying to deal with.

### Operating System

Linux

### Operating System Version

Arch Linux

### Installation method

Other

### Build Version

2024.12.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of GitHub. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by teidesu** [comment]
similar issue when using vscode with `Git: autofetch` enabled.

btw – the current ux of approving auth is overall not really good, primarily because the popup is inside the main application window. which also means that a) if the window is not open nothing happens, b) if the window is on another workspace – the workspace needs to be switched

for comparison, 1password's confirmation window looks like this, and appears on top of everything on the current workspace:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/11f02e2a-bc4b-4e7f-b9c5-2089f000f56f" />

**Comment by jtodddd** [comment]
Hi there,

I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below. 

Thanks!

**Comment by deify** [comment]
So I am having a similar issue with my git client application. Some transactions require multiple authentications. Also auto fetching is not working because of this. 
It would be nice to at least allow a certain application to reuse the ssh key while the session is still unlocked. This would be sufficient for my usecase.
The current approach requires to click way to many times for some simple tasks like fetching or pushing to a remote git instance with ssh

**Comment by svoop** [comment]
Apparently, Bitwarden requires authorization confirmation every time a SSH key is used since this happens with my Git client (Fork) as well and presumably it does on any such client. Maybe an authorization should be automatically re-given if the subsequen requests happen within a certain time frame (configurable in the settings)?

**Comment by schmitzel76** [comment]
I also see this issue in some automation scripts I have. These do call SCP multiple times, and each invocation requires an approval in the Bitwarden desktop app (on Windows in my case).

Bitwarden should not ask again within a certain time frame, at least not for the same application.

**Comment by lpsm-dev** [comment]
same issue in macos

**Comment by LexaNz** [comment]
Issue is the same if you use multiple jump-host to connect. 1 jump-host = 2 Confirmation prompt etc..

Would be nice to have a 'time' option, or a 'disable prompt' tick when key is used.

**Comment by daltonpearson** [comment]
I agree with @LexaNz. I too would like a 'time' option, or a 'disable prompt' tick box. It's annoying to have to refocus windows and click multiple times if I'm quickly ssh'ing into many boxes in a short period of time (updating packages, checking storage usage, other maintenance tasks, etc). I understand that it's a compromise but I also think that most people that use ssh keys for auth are capable of understanding the risks.

**Comment by 001123** [comment]
I have same, very bad UX.
Does that feature really think in real life usage case? Or just copy form 1 password with missing time concern about UX?
I think about buy premium plan, but I changed, 1password care about UX and it worth it!

**Comment by paulhey** [comment]
Yep, this is a frustrating feature at this point, especially when using VSCode and a ssh+git repo. The desktop client will periodically pop up over whatever I'm doing to prompt me to authorize again. The UI also defaults to highlighting the "X" in the top corner of the prompt, so if I happen to be typing something when it prompts, it cancels the authorization entirely. Might be better to have a toast notification or something similar instead?

**Comment by mrlindblom** [comment]
I have same issue. Add option to automatically authorize if bitwarden client is no locked. Authorization on every access is not need. 
However, i would like to have a notification when a key is used. Similar to how KeyAgent does it in windows.

**Comment by daltonpearson** [comment]
Looks like @quexten is implementing the very least a stop gap solution in this PR: https://github.com/bitwarden/clients/pull/13995
